code/lambda_handlers.py

```diff
@@ -0,0 +1,190 @@
+        raise ex
+
+
+def npv_handler(request, context):
+    """
+    Net Present Value of a cash flow series
+    :param request: Dict containing the parameters to pass to the formula.
+    :param context: Lambda execution context
+    :return: Dict with a 'result' entry containing the result of the calculation
+    """
+    schema = schemas.npv_schema
+
+    try:
+        validate(request, schema)
+        logger.info("NPV request: {}".format(request))
+        args = [request['rate'], request['values']]
+        logger.info("Calling numpy.npv with args: {}".format(args))
+        return {'result': numpy.npv(*args)}
+    except Exception as ex:
+        logger.error("Encountered an error while handling NPV request with args: {}. Exception: {}".format(request, ex))
+        raise ex
+
+
+def pmt_handler(request, context):
+    """
+    Compute the payment against loan principal plus interest
+    :param request: Dict containing the parameters to pass to the formula.
+    :param context: Lambda execution context
+    :return: Dict with a 'result' entry containing the result of the calculation
+    """
+    schema = schemas.pmt_schema
+
+    try:
+        validate(request, schema)
+        logger.info("PMT request: {}".format(request))
+        args = [request['rate'], request['nper'], request['pv'], request.get('fv', 0), request.get('type', 0)]
+        logger.info("Calling numpy.pmt with args: {}".format(args))
+        return {'result': numpy.pmt(*args)}
+    except Exception as ex:
+        logger.error("Encountered an error while handling PMT request with args: {}. Exception: {}".format(request, ex))
+        raise ex
+
+
+def ppmt_handler(request, context):
+    """
+    Compute the payment against loan principal
+    :param request: Dict containing the parameters to pass to the formula.
+    :param context: Lambda execution context
+    :return: Dict with a 'result' entry containing the result of the calculation
+    """
+    schema = schemas.ppmt_schema
+
+    try:
+        validate(request, schema)
+        logger.info("PPMT request: {}".format(request))
+        args = [request['rate'], request['per'], request['nper'], request['pv'], request.get('fv', 0), request.get('type', 0)]
+        logger.info("Calling numpy.ppmt with args: {}".format(args))
+        return {'result': numpy.ppmt(*args)}
+    except Exception as ex:
+        logger.error("Encountered an error while handling PPMT request with args: {}. Exception: {}".format(request, ex))
+        raise ex
+
+
+def irr_handler(request, context):
+    """
+    Internal Rate of Return calculation.
+    :param request: Dict containing the parameters to pass to the formula.
+    :param context: Lambda execution context
+    :return: Dict with a 'result' entry containing the result of the calculation
+    """
+    schema = schemas.irr_schema
+
+    try:
+        validate(request, schema)
+        logger.info("IRR request: {}".format(request))
+        args = [request['values']]
```

Comment:
Hmm interesting. In Excel, irr takes a second optional parameter called `guess` (default to 0.1 if omitted), but the numpy version doesn't... wonder if that's a problem...